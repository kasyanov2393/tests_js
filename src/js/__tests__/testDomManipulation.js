import Loader from '../loader';
import App from '../app';

const app = new App();

describe('DOM Manipulation testing', () => {
    beforeAll(() => {
        app.showApp();
    });

    afterAll(() => {
        app.hideApp();
    });

    it('should add element into #app block', () => {
        app.putHtmlInApp('some text');

        expect(app).toMatchSnapshot();
    });

    it('should show and hide loader element', () => {
        const loader = new Loader();

        const loaderId = '#loader';

        loader.showLoader();

        const queriedLoader = document.querySelector(loaderId);

        expect(queriedLoader).toBeDefined();
        expect(queriedLoader.style.display).toBe('block');
        expect(queriedLoader.textContent).toBe('Loading...');

        loader.hideLoader();

        expect(queriedLoader.style.display).toBe('none');
    });

    it('should call callbacks that added to buttons', () => {
        let buttonTemplates = '';
        const buttonClickHandler = jest.fn();

        for (let i = 0; i < 5; i += 1) {
            buttonTemplates = buttonTemplates.concat('<button></button>');
        }

        app.putHtmlInApp(buttonTemplates);

        const buttonElements = document.querySelectorAll('button');

        expect(buttonElements).not.toBeNull();
        expect(buttonElements.length).toBe(5);

        app.addEventListenersToButtons(buttonClickHandler);
        buttonElements.forEach((button) => { button.click(); });

        expect(buttonClickHandler).toBeCalled();
        expect(buttonClickHandler).toHaveBeenCalledTimes(5);
    });
});
