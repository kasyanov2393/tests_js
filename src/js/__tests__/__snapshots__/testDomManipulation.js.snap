// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DOM Manipulation testing should add element into #app block 1`] = `
App {
  "_element": <div
    id="app"
  >
    some text
  </div>,
}
`;
